contracts/modules/maker/MakerV2Invest.sol

```diff
@@ -0,0 +1,180 @@
+     * @return The exact amount of tokens that have been invested.
+     */
+    function addInvestment(
+        BaseWallet _wallet,
+        address _token,
+        uint256 _amount,
+        uint256 _period
+    )
+        external
+        onlyWalletOwner(_wallet)
+        onlyWhenUnlocked(_wallet)
+        returns (uint256 _invested)
+    {
+        require(_token == address(daiToken), "DM: token should be DAI");
+        joinDsr(_wallet, _amount);
+        _invested = _amount;
+        emit InvestmentAdded(address(_wallet), address(daiToken), _amount, _period);
+    }
+
+    /**
+     * @dev Exit invested postions.
+     * @param _wallet The target wallet.
+     * @param _token The token address.
+     * @param _fraction The fraction of invested tokens to exit in per 10000.
+     */
+    function removeInvestment(
+        BaseWallet _wallet,
+        address _token,
+        uint256 _fraction
+    )
+        external
+        onlyWalletOwner(_wallet)
+        onlyWhenUnlocked(_wallet)
+    {
+        require(_token == address(daiToken), "MV2: token not DAI");
+        require(_fraction <= 10000, "MV2: invalid fraction");
+        exitDsr(_wallet, _fraction);
+        emit InvestmentRemoved(address(_wallet), _token, _fraction);
+    }
+
+    /**
+     * @dev Get the amount of investment in a given token.
+     * @param _wallet The target wallet.
+     * @param _token The token address.
+     * @return The value in tokens of the investment (including interests) and the time at which the investment can be removed.
+     */
+    function getInvestment(
+        BaseWallet _wallet,
+        address _token
+    )
+        external
+        view
+        returns (uint256 _tokenValue, uint256 /* _periodEnd */)
+    {
+        if (_token == address(daiToken)) {
+            _tokenValue = dsrBalance(_wallet);
+        }
+    }
+
+    /* ****************************************** DSR wrappers ******************************************* */
+
+    /**
+    * @dev Grant access to the wallet's internal DAI balance in the VAT to an operator.
+    * @param _wallet The target wallet.
+    * @param _operator The grantee of the access
+    */
+    function grantVatAccess(BaseWallet _wallet, address _operator) internal {
+        if (vat.can(address(_wallet), _operator) == 0) {
+            invokeWallet(address(_wallet), address(vat), 0, abi.encodeWithSelector(VAT_HOPE, _operator));
+        }
+    }
+
+    function dsrBalance(BaseWallet _wallet) public view returns (uint256) {
+        return pot.chi().mul(pot.pie(address(_wallet))) / RAY;
+    }
+
+    /**
+    * @dev lets the owner deposit MCD DAI into the DSR Pot.
+    * @param _wallet The target wallet.
+    * @param _amount The amount of DAI to deposit
+    */
+    function joinDsr(
+        BaseWallet _wallet,
+        uint256 _amount
+    )
+        internal
+    {
+        if (daiToken.balanceOf(address(_wallet)) < _amount) {
```

Comment:
I'd say it all depends on how this is handled by the clients but I'll check with the clients that this is in fact useful. It may well be the case that we only allow DSR deposits from positive DAI balances anyways

Reply:
Confirmed with Miguel that this is not used and probably not useful. Will remove.